### ISSUE TITLE
Dropdown error when loaded and then later deleted from page

### ISSUE THREAD

**Issue by levimcg** [indiana-university/rivet-source#319]
## Describe the bug
cc: @illusivesunrae @zttodd This bug occurs in Rivet `1.7.2`.

This bug was reported on Slack. We were able to troubleshoot and come up with a temporary fix, this is something we should look into patching. From Slack conversation:

> My team has an issue with Rivet drop downs. The drop down element exists on the page, is opened, is closed, then the drop down element is deleted. The following clicks on the page result in this console error: "rivet.min.js:8 Uncaught TypeError: Cannot read property 'hasAttribute' of null". That happens because of the following code in the Rivet JavaScript closeMenu function:

```js
// Return if disabled dropdown is being closed programmatically
if (toggle.hasAttribute('disabled')) {
  return;
}
```

And a proposed solution:

> The Rivet drop down _handleKeydown function has the following code which would also solve the issue (it would reset activeMenu so there are no errors after the drop down menu is deleted):
```js
case KEYS.escape:
// If there's an open menu, close it.
if (activeMenu) {
  closeMenu(activeMenu);
}
if (activeToggle && activeToggle !== null) {
  activeToggle.focus();
}
/**
 * Resets the state variables so as not to interfere with other
 * Escape key handlers/interactions
 */
activeMenu = null;
```

## Expected behavior
Keydown handler should check for the existence of the dropdown toggle and/or menu and reset the `activeMenu` variable to null if either aren't on the page anymore.

## Additional context from Slack conversation:
> I only see three solutions without the Rivet source code changing
Solution 1:
Instead of calling window.Dropdown.init() once for all dropdowns, call window.Dropdown.init(context) and window.Dropdown.destroy(context) for every drop down we use. This solution would allow us to still use Rivet but is tedious and would break easily.
Solution 2:
Write our own custom drop down instead of using the Rivet drop down for situations Rivet has errors.
Solution 3:
The Rivet drop down _handleKeydown function has the following code which would also solve the issue (it would reset activeMenu so there are no errors after the drop down menu is deleted):

## Steps to reproduce
Add steps to reproduce the bug.

1. Add a dropdown menu to the DOM and initialize
1. Open and or close it
1. Remove it from the DOM (without re-initializing)
1. Click anywhere on the page, or use the escape key.

**Comment by eunmeeyi** [comment]
I've got same experiences and temporarily included rivet source code inside our application with adding 3 lines as follows. Please let me know this would occur any other issues. Hope it would be fixed soon.

```
    var toggle = document.querySelector('[' + TOGGLE_ATTR + '="' + id + '"]');
    // Add 3 lines below
    if ( !toggle ) {
       return;
    }

    // Return if disabled dropdown is being closed programmatically
    if (toggle.hasAttribute('disabled')) {
      return;
    }
```

**Comment by scottanthonymurray** [comment]
From closed duplicate issue #392:

![](https://user-images.githubusercontent.com/66126383/83267517-55807300-a192-11ea-861e-6879491a20b8.png)
